### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by vu-hoang-phan-form3**

### CODE REVIEW

**Review by samphillips-f3 — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,135 @@
...
+    # Location of the proxy's private key, if SSL is enabled
+    keyFilePath: "/etc/ssl/private/private.key"
+
+# Request forward proxy config
+proxy:
+  # URL where the proxy should forward the request to. It can be a server or another proxy.
+  upstreamTarget: "https://api.form3.tech/v1"
+  # Request signing config
+  signer:
+    # The key id stored on remote server that mapped to the public key
```
Comment (mszczygiel-form3) [inline README.md:51]:
```suggestion
    # The key id stored on remote server that maps to the public key
```

**Review by mszczygiel-form3 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,2 +1,98 @@
...
+In [example directory](./example), there are several files which help running the proxy locally:
+
+- [config_example.yaml](./example/config_example.yaml): A complete config file that should be in `--config` flag.
+- [cert.crt](./example/cert.crt): A self-signed certificate for an HTTPS proxy (TLS mode only).
+- [private.key](./example/private.key): The private key associated with the certificate above (TLS mode only).
+- [rsa_private_key.pem](./example/rsa_private_key.pem): A private key that is used to sign incoming request.
+- [rsa_public_key.pub](./example/rsa_public_key.pub): The public key associated with the private key above. It is not 
+  used by the proxy but will be useful if you need to verify the signature.
+- [docker-compose.yaml](./example/docker-compose.yaml): A convenient docker compose file that runs the proxy with all 
+  the config above. The proxy serves HTTPS requests at port 8080.
```
Comment (piotrolchawa-f3) [inline README.md:38]: We could mention the example `Dockerfile` here as a shortcut, so people can just jump to the file quickly

File: README.md [hunk 3]
```diff
@@ -1,2 +1,98 @@
...
+
+The proxy publishes certain metrics under `GET /-/prometheus` endpoint:
+
+|       Metric name        |   Type    | Description                                                                        |
+|:------------------------:|:---------:|------------------------------------------------------------------------------------|
+|   internal_error_total   |  Counter  | Total number of the proxy's internal errors. Upstream errors do not count.         |
+|   request_count_total    |  Counter  | Total number of requests coming to the proxy.                                      |
+|   signed_request_total   |  Counter  | Total number of incoming requests that have been signed and proxied.               |
+| signing_duration_seconds | Histogram | Request signing duration time in seconds.                                          |
+| request_duration_seconds | Histogram | Total request duration time in seconds, including signing and upstream processing. |
```
Comment (piotrolchawa-f3) [inline README.md:98]: I feel that we should keep good documentation standards for this OSS project and provide instructions for testing, development and contributing in the README file.

File: README.md (hunk 2 above)
Comment (samphillips-f3) [reply README.md:48]: added above

**Review by piotrolchawa-f3 — APPROVED** [review]

**Pull request merged**